addons/web/static/src/views/fields/dynamicplaceholder_hook.js

```diff
@@ -0,0 +1,78 @@
+/** @odoo-module **/
+
+import { usePopover } from "@web/core/popover/popover_hook";
+import { useModelField } from "@web/core/model_field_selector/model_field_hook";
+import { ModelFieldSelectorPopover } from "@web/core/model_field_selector/model_field_selector_popover";
+
+function useUniquePopover() {
+    const popover = usePopover();
+    let remove = null;
+    return Object.assign(Object.create(popover), {
+        add(target, component, props, options) {
+            if (remove) {
+                remove();
+            }
+            remove = popover.add(target, component, props, options);
+            return () => {
+                remove();
+                remove = null;
+            };
+        },
+    });
+}
+export function useDynamicplaceholder() {
+    const popover = useUniquePopover();
+    const modelField = useModelField();
+
+    let dynamicPlaceholderChain = [];
+
+    function update(chain) {
+        dynamicPlaceholderChain = chain;
+    }
+
+    return {
+        TRIGGER_KEY: '#',
+        /**
+         * Open a Model Field Selector which can select fields to create a dynamic
+         * placeholder string in the Input with or without a default text value.
+         *
+         * @public
+         * @param {HTMLElement} element
+         * @param {String} baseModel
+         * @param {function} validateCallback
+         * @param {function} closeCallback
+         * @param {function} onPositionCallback
+         */
+         async open(
```

Comment:
Could we have an options object rather than passing everything in separate args ? It makes the call site quite hard to understand IMO.